### PR TITLE
Align AsyncGRPO clip-ratio metrics with GRPOTrainer

### PR DESCRIPTION
**Pull request by qgallouedec**
`AsyncGRPOTrainer` logs a `ratio` metric (no `GRPOTrainer` equivalent) and a single, advantage-sign-agnostic `clip_ratio`. `GRPOTrainer` instead gates clipping on the advantage sign and logs `clip_ratio/low_mean`, `clip_ratio/high_mean`, `clip_ratio/region_mean` (+ `low_min` / `high_max`). Aligning means changing what is computed and the logged metric keys.

---

> [!NOTE]
> **Low Risk**
> Observability-only change to metric definitions and keys; the GRPO loss path is not modified.
> 
> **Overview**
> **AsyncGRPO** training logs now match **`GRPOTrainer`** for PPO clip saturation instead of a single `clip_ratio` that ignored advantage sign.
> 
> Clip counts only treat a token as clipped when clipping is **binding in the policy-relevant direction**: low clip when `coef_1` is below the floor and advantage is negative, high clip when above the ceiling and advantage is positive. Logged keys become `clip_ratio/low_mean`, `clip_ratio/high_mean`, `clip_ratio/region_mean`, plus cross-rank `clip_ratio/low_min` and `clip_ratio/high_max` via `accelerator.gather` and `nanmin`/`nanmax`. The distributed stats all-reduce vector is extended accordingly; **`ratio`**, **kl**, and **entropy** metrics are unchanged.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 33613db246354f8a23e1c2bb9f3617495f4dee62. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/async_grpo/async_grpo_trainer.py [hunk 1]
```diff
@@ -586,21 +586,66 @@ def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=N
...
+                global_region_clip_sum,
+                global_count,
+            ) = stats.unbind(0)
             self._metrics["train"]["ratio"].append((global_ratio_sum / global_count).item())
             self._metrics["train"]["kl"].append((global_kl_sum / global_count).item())
             self._metrics["train"]["entropy"].append((global_entropy_sum / global_count).item())
-            self._metrics["train"]["clip_ratio"].append((global_clip_sum / global_count).item())
+            self._metrics["train"]["clip_ratio/low_mean"].append((global_low_clip_sum / global_count).item())
+            self._metrics["train"]["clip_ratio/high_mean"].append((global_high_clip_sum / global_count).item())
+            self._metrics["train"]["clip_ratio/region_mean"].append((global_region_clip_sum / global_count).item())
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/experimental/async_grpo/async_grpo_trainer.py:701]: Correct, when per-rank token counts differ, the global-sum mean (Σ clipped / Σ count) and GRPO's gather + nanmean of per-rank fractions disagree. But this isn't specific to clip_ratio: AsyncGRPO already aggregates ratio/kl/entropy the same global-sum way, so the means here just follow the house style. Switching only clip_ratio to nanmean would make it inconsistent with the rest of the function.

And global-sum is the correct one!: GRPO's unweighted nanmean over ranks over-weights ranks with fewer tokens. So the right fix is to move GRPO/RLOO to global-sum, not to bend AsyncGRPO to match.

File: trl/experimental/async_grpo/async_grpo_trainer.py [hunk 2]
```diff
@@ -643,21 +643,66 @@ def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=N
...
             )
 
-            # Batch all-reduce: [ratio_sum, kl_sum, entropy_sum, clip_sum, count]
-            stats = torch.stack([local_ratio_sum, local_kl_sum, local_entropy_sum, local_clip_sum, local_count])
+            # Per-rank clip fractions, gathered below to report the cross-rank saturation extrema.
+            local_low_clip_mean = local_low_clip_sum / local_count.clamp(min=1.0)
+            local_high_clip_mean = local_high_clip_sum / local_count.clamp(min=1.0)
+
+            # Batch all-reduce: [ratio_sum, kl_sum, entropy_sum, low_clip_sum, high_clip_sum, region_clip_sum, count]
+            stats = torch.stack(
```
Comment (AmineDiro) [inline trl/experimental/async_grpo/async_grpo_trainer.py:675]: super cool to have these stats 👍🏼

**Review by AmineDiro — APPROVED** [review]
